modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/load/IgniteCacheRandomOperationBenchmark.java

```diff
@@ -465,15 +545,64 @@ private void executeRandomOperation(IgniteCache cache) throws Exception {
 
             case SQL_QUERY:
                 doSqlQuery(cache);
+                break;
+
+            case CONTINUOUS_QUERY:
+                doContinuousQuery(cache, map);
         }
+        storeStatistics(cache.getName(), map, op);
+    }
+
+    /**
+     * @param cacheName Ignite cache name.
+     * @return Operation.
+     */
+    @NotNull private Operation nextRandomOperation() {
+        return allowOperations.get(nextRandom(allowOperations.size()));
+    }
+
+    /**
+     * @param cacheName Ignite cache name.
+     * @param map Parameters map.
+     * @param op Operation.
+     */
+    private void storeStatistics(String cacheName, Map<Object, Object> map, Operation op) {
+        String opCacheKey = String.format("%s_%s", op, cacheName);
+        Long opAmount = (Long)map.get("amount");
+        opAmount = opAmount == null ? 1L : opAmount + 1;
+        Integer opCacheCnt = (Integer)map.get(opCacheKey);
+        opCacheCnt = opCacheCnt == null ? 1 : opCacheCnt + 1;
+
+        if (opAmount % 100 == 0)
+            updateStat(map);
+        else
+            map.put(opCacheKey, opCacheCnt);
+
+        map.put("amount", opAmount);
```

Comment:
The whole operation is not atomic since other Thread can update the same key in parallel.